include/ptp_clock.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _PTP_CLOCK_H_
+#define _PTP_CLOCK_H_
+
+#include <stdint.h>
+#include <device.h>
+#include <misc/util.h>
+#include <net/ptp_time.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * Special alignment is needed for ptp_clock which is stored in
+ * a ptp_clock linker section if there are more than one ptp clock
+ * in the system. If there is only one ptp clock,
+ * then this alignment is not needed, unfortunately this cannot be
+ * known beforehand.
+ *
+ * The ptp_clock struct needs to be aligned to 32 byte boundary,
+ * otherwise the __ptp_clock_end will point to wrong location and ptp_clock
+ * initialization done in ptp_clock_init() will not find proper values
+ * for the second interface.
+ *
+ * So this alignment is a workaround and should eventually be removed.
+ */
+#define __ptp_clock_align __aligned(32)
+
+struct ptp_clock;
+
+struct ptp_clock_driver_api {
+	int (*set)(struct ptp_clock *clk, struct net_ptp_time *tm);
```

Comment:
Actually that's a no-go. As a device driver API, the mandated rule (unexplained anywhere) is to have struct device * as first parameter. 

Well, it's an easy change: just use *dev param, and for all inlines below, get the api from dev etc...